docs/concepts/workloads/controllers/replicaset.md

```diff
@@ -93,7 +93,97 @@ frontend-dnjpy   1/1       Running   0          1m
 frontend-qhloh   1/1       Running   0          1m
 ```
 
-## ReplicaSet as an Horizontal Pod Autoscaler target
+## Writing a ReplicaSet Spec
+
+As with all other Kubernetes config, a ReplicaSet needs the `apiVersion`, `kind`, and `metadata` fields.  For
+general information about working with config files, see [here](/docs/user-guide/simple-yaml/),
+[here](/docs/user-guide/configuring-containers/), and [here](/docs/concepts/tools/kubectl/object-management-overview/).
+
+A ReplicaSet also needs a [`.spec` section](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status).
+
+### Pod Template
+
+The `.spec.template` is the only required field of the `.spec`. The `.spec.template` is a 
+[pod template](/docs/concepts/workloads/pods/pod-overview/#pod-templates). It has exactly the same schema as a 
+[pod](/docs/concepts/workloads/pods/pod/), except that it is nested and does not have an `apiVersion` or `kind`.
+
+In addition to required fields of a pod, a pod template in a ReplicaSet must specify appropriate
+labels and an appropriate restart policy.
+
+For labels, make sure to not overlap with other controllers. For more information, see [pod selector](#pod-selector). 
+For restart policy, only a  [`.spec.template.spec.restartPolicy`](/docs/concepts/workloads/pods/pod-lifecycle/) 
+equal to `Always` is allowed, which is the default if not specified.
+
+For local container restarts, ReplicaSet delegate to an agent on the node,
+for example the [Kubelet](/docs/admin/kubelet/) or Docker.
+
+### Pod Selector
```

Comment:
You should consider adding a paragraph that indicates that, in Kuberentes 1.8 and later, selectors are no longer defaulted and are immutable after creation.

Reply:
Done.